### ISSUE TITLE
Bug: Chaining Function Calls broken?

### ISSUE THREAD

**Issue by borislopezaraoz** [google/python-fire#150]
This code from docs (simplified):

```
import fire

class Airport(object):
    def __init__(self, code):
        self.code = code

if __name__ == '__main__':
    fire.Fire(Airport)

```

Produces this error when trying to chain upper function (command also from docs):

```
python example.py --code=ALB code upper
Fire trace:
1. Initial component
2. Instantiated class "Airport" (example.py:3)
3. Accessed property "code"
4. Accessed property "upper"
5. ('The function received no value for the required argument:', 'self')

Type:        builtin_function_or_method
String form: <built-in method upper of str object at 0x1063f2a40>
Docstring:   Return a copy of the string converted to uppercase.

Usage:       example.py --code=ALB code upper SELF
             example.py --code=ALB code upper --self SELF
```

fire==0.1.3

**Comment by dbieber** [comment]
Hmm, a quick check on my system shows `python example.py --code=ALB code upper` is correctly producing `ALB` for both Python 2.7 (ipython==5.5.0) and 3.5.4rc1 (ipython==7.2.0) for fire==1.3.0.

What version of Python are you on? Is ipython installed, and if so at what version? OS?

**Comment by dbieber** [comment]
Fixed by 668007ae41391f5964870b4597e41493a936a11e.
